x-pack/plugins/secops/public/components/page/hosts/events_table/index.tsx

```diff
@@ -201,7 +202,8 @@ export const formatIpSafely = (path: string, data: Ecs): JSX.Element => {
     if (txt === '') {
       return getEmptyStringTag();
     } else {
-      return txt && txt.slice ? <>{txt.slice(0, 45)}</> : <>{txt}</>;
+      const ip = txt && txt.slice ? txt.slice(0, 45) : txt;
```

Comment:
It's nice to see that we are filtering sensitive information here! 👍I've got a question about the length, would like to know how do we know it is exactly 45 chars we need here?

Another question is the return type of `formatIpSafely`. What's the difference between `React. ReactNode` and `JSX.Element`?

Reply:
So the length-check actually comes from this recent PR https://github.com/elastic/kibana/pull/31741#discussion_r259478208 -- with `45` characters being the max-length of an IP we'll display (lengthiest case being an IPv4 mapped to an IPv6).

As for the difference between `React.ReactNode` and `JSX.Element`. My understanding is `JSX.Element` is any object that conforms to the `JSX.Element` interface (having a `type`, `props`, and `key`) and `React.ReactNode` is the set of all possible return values for a component. Since it looks like `formatIpSafely()` returns either a `ReactFragment` or `ReactComponent`, the return type is okay as `JSX.Element`, but had it returned just text, it would need to be `React.ReactNode`. 

I hope this was helpful, and thanks for asking, as this gave me a reason to dig around the React types a bit more. :) It's interesting, `JSX.Element` just extends `ReactElement<any>` whereas `ReactNode` is practically everything under the sun! (`ReactChild | ReactFragment | ReactPortal | string | number | boolean | null | undefined`).

Reply:
nit: This would look better and be smaller if it was just `(isString(txt) || isArray(txt)) ? txt.slice(0, 45) : txt`

No changes, just pointing out checking for the existence of a function is less semantic. As it stands it does look like the question remains can that be an array and a string? And if so, are our GraphQL typings correct?